.pre-commit-config.yaml

```diff
@@ -16,6 +16,16 @@ repos:
       - id: mixed-line-ending
       - id: end-of-file-fixer
       - id: trailing-whitespace
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.3.1
+    hooks:
+      - id: pyupgrade
+        args: ["--py37-plus"]
+  - repo: https://github.com/akaihola/darker
+    rev: 1.6.1
+    hooks:
+      - id: darker
+        additional_dependencies: [black]
```

Comment:
This is awesome @dbast !

Thanks for working on it.

How about using https://github.com/charliermarsh/ruff#usage ? whcih is faster and bundles a bunch of tools into a single one?

Reply:
ruff would replace flake8? something we could try via another PR.

the idea here is to apply pyupgrade (don't see its functionality included in ruff) and to also style all the lines that are upgraded by it with black, which is done by darker (gradually styles lines with black that are anyway already changed). sorry if that wasn't clear.